src/solutions/optimization_solutions.jl

```diff
@@ -17,25 +17,25 @@ Representation of the solution to an nonlinear optimization defined by an Optimi
   callback (`sol.retcode === :Terminated`), or whether it exited due to an error. For more
   details, see the return code section of the DifferentialEquations.jl documentation.
 """
-struct OptimizationSolution{T, N, uType, P, A, Tf, O} <: AbstractOptimizationSolution{T, N}
+struct OptimizationSolution{T, N, uType, C, A, Tf, O} <: AbstractOptimizationSolution{T, N}
     u::uType # minimizer
-    prob::P # optimization problem
+    cache::C # optimization cache
```

Comment:
I'm not so sure about this, since that is an interface break. We'd need to do that to all solution types.

Reply:
`LinearSolution` does this already
https://github.com/SciML/SciMLBase.jl/blob/b655d0657f9aab56a7619aa0584c2ef569d670f6/src/solutions/basic_solutions.jl#L28

But we can stick the cache definition in here or an abstract cache type and dispatch on it and leave the method name as it is.

Reply:
interesting, okay then go for it. It might have some memory consequences though?